optimizely/decision_service.py

```diff
@@ -489,3 +489,63 @@ def get_variation_for_feature(self, project_config, feature, user_context, optio
         if rollout_variation_reasons:
             decide_reasons += rollout_variation_reasons
         return variation, decide_reasons
+
+    def validated_forced_decision(self, decision_context, user_context):
+        """
+        Gets forced decisions based on flag key, rule key and variation.
+
+        Args:
+            decision context: a decision context
+            user_context context: a user context
+
+        Returns:
+            Variation of the forced decision.
+        """
+        reasons = []
+
+        forced_decision = user_context.find_forced_decision(decision_context)
+
+        flag_key = decision_context.flag_key
+        rule_key = decision_context.rule_key
+
+        if forced_decision:
+            # we use config here so we can use get_flag_variation() function which is defined in project_config
+            # otherwise we would us user_context.client instead of config
+            config = user_context.client.config_manager.get_config() if user_context.client else None
+            if not config:
+                return None, reasons
+            variation = config.get_flag_variation(flag_key, 'key', forced_decision.variation_key)
+            if variation:
+                if rule_key:
+                    user_has_forced_decision = enums.ForcedDecisionLogs \
```

Comment:
please update this name to something that suggests it is a reason.
`user_has_forced_decision`